v2/go.mod

```diff
@@ -0,0 +1,11 @@
+module github.com/nats-io/jwt/v2
+
+require (
+	github.com/nats-io/jwt v0.3.2
+	github.com/nats-io/nkeys v0.1.4
+	github.com/stretchr/testify v1.5.1
+)
+
+replace github.com/nats-io/jwt v0.3.2 => ../
```

Comment:
Not sure how well this will work since it is an '[inter dependent module](https://github.com/golang/go/issues/27542)' but maybe it will... think this more or less following the [major subdirectory strategy](https://github.com/go-modules-by-example/index/tree/master/016_major_version_repo_strategy#major-subdirectory-strategy). The other alternative is to follow the [major branch strategy](https://github.com/go-modules-by-example/index/tree/master/016_major_version_repo_strategy#major-branch-strategy) where can skip having a v2 folder and master with only the v2 contents.  I think also requires making the v1.0.0 tag, but can then continue development of both v1 and v2 series just in different branches.

Reply:
we need to be able to read/write v1 from v2, if we run into a problem while developing that portion, then we can change on that.

Reply:
yes I know that, but we need to make sure that we are taking the right strategy to avoid causing issues to other repos in the ecosystem

Reply:
Not following the other issues on the nats side - the JWT library has always been a go.mod library since day one, so all the current clients are already importing the lib as such, regardless of whether the other products rely on vendoring to do their builds.

The [subdirectory strategy is how they say to use modules](https://blog.golang.org/using-go-modules), not some invention. The interdependent is a gray area, but in all honesty, I think they are worried about replacements that cannot be honored. In our case, it is the same repo.

Now if we want to do flat (perfectly ok), we need to take a stance and do it. This also means that the library will probably duplicate a bunch of types, but I am sure we can sort it out.